src/main/java/frc/robot/subsystems/arm/ArmSystemModel.java

```diff
@@ -0,0 +1,161 @@
+package frc.robot.subsystems.arm;
+
+import frc.robot.utils.motors.FalconConstants;
+import org.ejml.data.MatrixType;
+import org.ejml.simple.SimpleMatrix;
+
+/**
+ * This class is used to calculate the feedforward for the arm.
+ * Notice that this class does not calculate velocities or accelerations,
+ * and relies on the user to do so (in the subsystem class).
+ *
+ * Documentation for the math used in this class can be found here:
+ * https://www.chiefdelphi.com/t/whitepaper-two-jointed-arm-dynamics/423060?u=dan
+ */
+public class ArmSystemModel {
+    private final double m1;
+    private final double m2;
+    private final double l1;
+    private final double r1;
+    private final double r2;
+    private final double I1;
+    private final double I2;
+    private final double g1;
+    private final double g2;
+
+    private final SimpleMatrix M_MATRIX = new SimpleMatrix(2, 2, MatrixType.DDRM);
+    private final SimpleMatrix C_MATRIX = new SimpleMatrix(2, 2, MatrixType.DDRM);
+    private final SimpleMatrix Tg_VECTOR = new SimpleMatrix(2, 1, MatrixType.DDRM);
+    private final SimpleMatrix Kb_MATRIX = new SimpleMatrix(2, 2, MatrixType.DDRM);
+    private final SimpleMatrix B_MATRIX = new SimpleMatrix(2, 2, MatrixType.DDRM);
+
+    /**
+     * The constructor for the system model.
+     *
+     * @param systemConstants the system constants of the arm.
+     */
+    public ArmSystemModel(SystemConstants systemConstants) {
+        var shoulderJointConstants = systemConstants.getShoulderJointConstants();
+        var elbowJointConstants = systemConstants.getElbowJointConstants();
+        this.m1 = shoulderJointConstants.mass;
+        this.m2 = elbowJointConstants.mass;
+        this.l1 = shoulderJointConstants.length;
+        this.r1 = shoulderJointConstants.cmRadius;
+        this.r2 = elbowJointConstants.cmRadius;
+        this.I1 = shoulderJointConstants.momentOfInertia;
+        this.I2 = elbowJointConstants.momentOfInertia;
+        this.g1 = shoulderJointConstants.gearing;
+        this.g2 = elbowJointConstants.gearing;
+        int n1 = shoulderJointConstants.numberOfMotors;
+        int n2 = elbowJointConstants.numberOfMotors;
+
+        B_MATRIX.set(0, 0, g1 * n1 * FalconConstants.Kt / FalconConstants.R);
+        B_MATRIX.set(1, 1, g2 * n2 * FalconConstants.Kt / FalconConstants.R);
+        B_MATRIX.set(1, 0, 0);
+        B_MATRIX.set(0, 1, 0);
+
+        Kb_MATRIX.set(0, 0, g1 * g1 * n1 * FalconConstants.Kt / FalconConstants.R / FalconConstants.Kv);
+        Kb_MATRIX.set(1, 1, g2 * g2 * n2 * FalconConstants.Kt / FalconConstants.R / FalconConstants.Kv);
```

Comment:
I'd advise against using the `FalconConstants` here, which I guess originate from the manufacturer's data. Each motor _unit_ is a bit different, and the exact gains should be calculated using SysId.

Reply:
What's SysId?